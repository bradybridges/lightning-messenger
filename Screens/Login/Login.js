import React, { Component } from 'react';
import { Text,  
  View, 
  TextInput, 
  Button, 
  StyleSheet, 
  Dimensions, 
  KeyboardAvoidingView,
  TouchableOpacity,
  ImageBackground,
  Image,
} from 'react-native';
import * as firebase from 'firebase';
import * as Constants from '../../Constants/Constants';
import 'firebase/auth';
import ApiKeys from '../../ApiKeys';
const logo = require('../../assets/logo.png');


export default class Login extends Component {
  state = {
    email: '',
    password: '',
  }

  componentDidMount = () => {
    if(!firebase.apps.length) {
      firebase.initializeApp(ApiKeys.FirebaseConfig);
    } 
    firebase.auth().onAuthStateChanged(user => {
      if(user) {
        this.props.navigation.replace('Home');
      }
    });
  }

  handleEmailChange = (email) => {
    this.setState({ email });
  }

  handlePasswordChange = (password) => {
    this.setState({ password });
  }

  handleLogin = async () => {
    const { email, password } = this.state;
    await firebase.auth().signInWithEmailAndPassword(email, password)
      .catch((error) => {
        this.setState({ error });
        alert(error.message);
        alert(error.code);
      });
  }

  render() {
    const { navigate } = this.props.navigation;
    return (
      <View style={styles.container}>
        <Image source={logo} style={{ width: 150, height: 150 }}/>
        <Text style={styles.header}>Lightning Messenger</Text>
        <KeyboardAvoidingView style={styles.loginContainer}>
          <Text style={styles.label}>Email</Text>
          <TextInput
            style={styles.input}
            value={this.state.email}
            onChangeText={this.handleEmailChange}
          />
          <Text style={styles.label}>Password</Text>
          <TextInput
            style={styles.input}
            secureTextEntry={true}
            value={this.state.password}
            onChangeText={this.handlePasswordChange}
          />
          <View style={styles.buttonContainer}>
            <TouchableOpacity style={styles.button} onPress={this.handleLogin}><Text style={styles.buttonText}>Login</Text></TouchableOpacity>
            <TouchableOpacity style={styles.button} onPress={() => navigate('CreateAccount')}><Text style={styles.buttonText}>Create Account</Text></TouchableOpacity>
          </View>

          </KeyboardAvoidingView>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'black',
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  input: {
    width: Dimensions.get('window').width * .9,
    maxWidth: 400,
    height: 60,
    backgroundColor: 'white',
    borderRadius: 4,
    fontSize: 24,
    textAlign: 'center',
  },
  label: {
    color: 'white',
    fontSize: 24,
  },
  loginContainer: {
    height: 400,
    display: 'flex',
    justifyContent: 'space-evenly',
    alignItems: 'center',
  },
  buttonContainer: {
    width: Dimensions.get('window').width * .8,
    height: 140,
    display: 'flex',
    justifyContent: 'space-evenly',
  },  
  header: {
    color: 'white',
    fontSize: 40,
  },
  button: { 
    backgroundColor: 'white', 
    height: 50, 
    display: 'flex', 
    alignItems: 'center', 
    justifyContent: 'center',
    borderRadius: 4,
  },
  buttonText: {
    fontSize: 24,
  }
});

