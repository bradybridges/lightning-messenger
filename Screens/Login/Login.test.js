import React from 'react';
import { shallow } from 'enzyme';
import Login from './Login';

describe('Login', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Login navigation={jest.fn()} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have properties of email, password and loading in state', () => {
    expect(wrapper.state()).toEqual({ email: '', password: '', loading: false });
  });

  it('handleEmailChange should update email in state', () => {
    wrapper.instance().handleEmailChange('test@test.com');
    expect(wrapper.state('email')).toEqual('test@test.com');
  });

  it('handlePasswordChange should update password in state', () => {
    wrapper.instance().handlePasswordChange('password');
    expect(wrapper.state('password')).toEqual('password');
  });
});
