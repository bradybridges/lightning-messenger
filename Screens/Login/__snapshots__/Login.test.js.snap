// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#14272E",
      "flex": 1,
      "justifyContent": "flex-start",
      "paddingTop": 20,
    }
  }
>
  <StatusBar
    animated={false}
    barStyle="light-content"
    showHideTransition="fade"
  />
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 40,
        "marginVertical": 10,
      }
    }
  >
    Login
  </Text>
  <KeyboardAvoidingView
    enabled={true}
    keyboardVerticalOffset={0}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": 400,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Email"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 4,
          "fontSize": 24,
          "height": 60,
          "marginVertical": 10,
          "maxWidth": 400,
          "textAlign": "center",
          "width": 675,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Password"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 4,
          "fontSize": 24,
          "height": 60,
          "marginVertical": 10,
          "maxWidth": 400,
          "textAlign": "center",
          "width": 675,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "width": 600,
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#121716",
            "borderColor": "#6E8F76",
            "borderRadius": 4,
            "borderWidth": 1,
            "display": "flex",
            "height": 50,
            "justifyContent": "center",
            "marginTop": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#6E8F76",
              "fontSize": 24,
            }
          }
        >
          Login
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#121716",
            "borderColor": "#6E8F76",
            "borderRadius": 4,
            "borderWidth": 1,
            "display": "flex",
            "height": 50,
            "justifyContent": "center",
            "marginTop": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#6E8F76",
              "fontSize": 24,
            }
          }
        >
          Join
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#121716",
            "borderColor": "#6E8F76",
            "borderRadius": 4,
            "borderWidth": 1,
            "display": "flex",
            "height": 50,
            "justifyContent": "center",
            "marginTop": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#6E8F76",
              "fontSize": 24,
            }
          }
        >
          Reset Password
        </Text>
      </TouchableOpacity>
    </View>
  </KeyboardAvoidingView>
  <ActivityIndicator
    animating={false}
    color="#6E8F76"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "position": "absolute",
        "top": "30%",
        "zIndex": 5,
      }
    }
  />
</View>
`;
